cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/BuildCommand.java

```diff
@@ -198,6 +198,10 @@ public BuildCommand() {
             "generation")
     private String graalVMBuildOptions;
 
+    @CommandLine.Option(names = "--optimize-memory", hidden = true,
```

Comment:
I prefer to use `--optimize-dependency-compilation` flag here. It clearly explains the intent without exposing internal details. 

Description: "[EXPERIMENTAL] Enables memory-efficient compilation of package dependencies using separate processes. This can help prevent out-of-memory issues during initial compilation with a clean central cache."

Reply:
Do you mean we should add it to the help text too? Was thinking of keeping it hidden.

Reply:
The flag is changed in d259d32. Since this is a new addition, thought we could introduce the `[EXPERIMENTAL]...` in U10 and keep hidden in 9.x. WDYT?

Reply:
Yes. Shall we add the description I suggested? @azinneera

Reply:
Fixed with 1708ad3